extension/main.js

```diff
@@ -55,12 +55,17 @@
 	function handleNotificationClicked(notificationId) {
 		const url = window.GitHubNotify.settings.get(notificationId);
 		if (url) {
-			fetch(url).then(res => res.json()).then(json => {
-				openTab(json.html_url);
+			window.GitHubNotify.request(url).then(res => res.json()).then(json => {
+				if (json.message === 'Not Found') {
+					openTab(window.GitHubNotify.getTabUrl());
+				} else {
+					openTab(json.html_url);
+				}
```

Comment:
well I didn't liked this kind of verbose branching just to figure out which url to pass, and I don't really like doing ternary in function call either. As of variable: V8 will probably just optimize it away.